cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -114,6 +121,40 @@ protected void startUp() throws Exception {
     runningThread = Thread.currentThread();
   }
 
+  private void checkSuspended() {
```

Comment:
so this is really a ``waitWhileSuspended()``, right? and not just ``checkSuspended``?